src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityCodeFixProvider.cs

```diff
@@ -0,0 +1,169 @@
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+
+            var inconsistentAccessibilityInfo = await GetInconsistentAccessibilityInfoAsync(context.Document, diagnostic, context.CancellationToken).ConfigureAwait(false);
+
+            var typeLocation = await FindTypeLocationInSourceCodeAsync(context.Document, inconsistentAccessibilityInfo.TypeToChangeAccessibility, context.CancellationToken).ConfigureAwait(false);
+
+            if(typeLocation != Location.None)
+            {
+                context.RegisterCodeFix(CodeAction.Create(inconsistentAccessibilityInfo.CodeActionMessage, c => ChangeTypeAccessibilityAsync(context.Document.Project.Solution, inconsistentAccessibilityInfo.NewAccessibilityModifiers, typeLocation, c)), diagnostic);
+            }
+        }
+
+        private static async Task<InconsistentAccessibilityInfo> GetInconsistentAccessibilityInfoAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            InconsistentAccessibilityInfoProvider inconsistentAccessibilityProvider = null;
+
+            switch (diagnostic.Id)
+            {
+                case InconsistentAccessibilityInMethodParameterCompilerErrorNumber:
+                    inconsistentAccessibilityProvider = new InconsistentAccessibilityInMethodParameter();
+                    break;
+            }
+
+            return await inconsistentAccessibilityProvider.GetInconsistentAccessibilityInfoAsync(document, diagnostic, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<Location> FindTypeLocationInSourceCodeAsync(Document document, TypeSyntax type, CancellationToken cancellationToken)
+        {
+            var result = Location.None;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetSymbolInfo(type, cancellationToken).Symbol;
+
+            if(typeSymbol != null)
+            {
+                var locationInSource = typeSymbol.Locations.FirstOrDefault(location => location.IsInSource);
+                if(locationInSource != null)
+                {
+                    result = locationInSource;
+                }
+            }
+
+            return result;
+        }
+
+        private static async Task<Document> ChangeTypeAccessibilityAsync(Solution solution, SyntaxTokenList newAccessibilityModifiers, Location typeLocation, CancellationToken cancellationToken)
+        {
+            var document = solution.GetDocument(typeLocation.SourceTree);
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var result = document;
+
+            var declaration = syntaxRoot.FindNode(typeLocation.SourceSpan) as MemberDeclarationSyntax;
+            if (declaration != null)
```

Comment:
When we get to the code fix provider we have to be sure of what syntax we are dealing with. There should not be any situation where `SyntaxRoot.FindNode(typeLocation.SourceSpan)` is not a `MemberDeclarationSyntax`.
I suggest you change this to `(MemberDeclarationSyntax)SyntaxRoot.FindNode(typeLocation.SourceSpan)` and remove the null check.